include/revng/Support/YAMLTraits.h

```diff
@@ -273,3 +288,75 @@ llvm::Expected<T> deserializeFileOrSTDIN(const llvm::StringRef &Path) {
   return deserialize<T>((*MaybeBuffer)->getBuffer());
 }
 // clang-format on
+
+template<HasScalarOrEnumTraits... T>
+requires(sizeof...(T) > 0) struct llvm::yaml::ScalarTraits<std::variant<T...>> {
```

Comment:
As, discussed, let's avoid this. Let's split the `Rank` needing this in revng-c in two distinct ranks.

Reply:
Removed the trait for `std::variant`